### DISCUSSION TITLE
Error in TID

### DISCUSSION THREAD

**Discussion by duartefontes** [FeynCalc/feyncalc#123] [Q&A]
Hi,
if I define:
t0=DiracTrace[GAD[-J1].GSD[k1].GSD[-k1+q1].GAD[-J3].GSD[-k1+p1-q2].GSD[-k1+p1].GA[5]]*FAD[{-k1,mb}]*FAD[{-k1+p1,mb}]*FAD[{-k1+q1,mt}]*FAD[{-k1+p1-q2,mt}],
and then perform TID[t0, k1], I obtain the error:
"Error! TID has encountered a fatal problem and must abort the computation. The problem reads: Running tidSingleIntegral failed to achieve full tensor reduction of the unique integrals in".

(Note: the message really ends like this, in an incomplete fashion). I checked that I do not get the error if I perform TID[t0 // DiracSimplify, k1].

What is happening here? Is there any way to avoid the error in TID[t0, k1]?

Thank you in advance,
Duarte

**Comment by vsht** [comment]
Hi,

you have a trace with 6 Dirac matrices and a Gamma 5 in NDR, which is precisely a quantity that FeynCalc
cannot evaluate automatically. `TID[t0 // DiracSimplify, k1]` works because the first application of DiracSimplify
simplifies the chain inside the trace (but without computing the trace), while the second one again calls DiracTrace
and uses the linearity of the trace (again without computing it).

I don't see any good strategy for DiracSimplify here, since usually such traces are evaluated in the HV scheme, as
the naive scheme is completely ambiguous for such quantities. The philosophy is that the user should handle such
objects accordingly before passing them to FeynCalc, e.g. via something like

```
t1 = t0 /.  DiracTrace[x_] :>    FCTraceExpand[DiracTrace[DiracSimplify[x, InsideDiracTrace -> True]]]
TID[t1, k1, FCVerbose -> 1]
```

**Reply by duartefontes** [reply]
Dear Vlad,
thanks a lot.
Duarte
